internal/promotion/runner/builtin/argocd_updater_test.go

```diff
@@ -1081,29 +1078,100 @@ func Test_argoCDUpdater_syncApplication(t *testing.T) {
 					},
 				},
 			},
-			assertions: func(t *testing.T, err error) {
+			assertions: func(t *testing.T, err error, patched *argocd.Application) {
 				require.NoError(t, err)
 			},
 		},
+		{
+			name: "updates Sources when present",
+			app: &argocd.Application{
+				ObjectMeta: metav1.ObjectMeta{Name: "fake", Namespace: "fake"},
+				Spec: argocd.ApplicationSpec{
+					Sources: argocd.ApplicationSources{{TargetRevision: "old-rev"}},
+				},
+			},
+			desiredSources: argocd.ApplicationSources{{TargetRevision: "new-rev"}},
+			assertions: func(t *testing.T, err error, patched *argocd.Application) {
+				require.NoError(t, err)
+				require.NotNil(t, patched)
+				require.Len(t, patched.Spec.Sources, 1)
+				require.Equal(t, "new-rev", patched.Spec.Sources[0].TargetRevision)
+			},
+		},
+		{
+			name: "updates Source when Sources not present",
+			app: &argocd.Application{
+				ObjectMeta: metav1.ObjectMeta{Name: "fake", Namespace: "fake"},
+				Spec: argocd.ApplicationSpec{
+					Source: &argocd.ApplicationSource{TargetRevision: "old-rev"},
+				},
+			},
+			desiredSources: argocd.ApplicationSources{{TargetRevision: "new-rev"}},
+			assertions: func(t *testing.T, err error, patched *argocd.Application) {
+				require.NoError(t, err)
+				require.NotNil(t, patched)
+				require.NotNil(t, patched.Spec.Source)
+				require.Equal(t, "new-rev", patched.Spec.Source.TargetRevision)
+			},
+		},
+		{
+			name: "updates Sources when both Source and Sources are present",
+			app: &argocd.Application{
+				ObjectMeta: metav1.ObjectMeta{Name: "fake", Namespace: "fake"},
+				Spec: argocd.ApplicationSpec{
+					Source:  &argocd.ApplicationSource{TargetRevision: "old-rev-source"},
+					Sources: argocd.ApplicationSources{{TargetRevision: "old-rev-sources"}},
+				},
+			},
+			desiredSources: argocd.ApplicationSources{{TargetRevision: "new-rev"}},
+			assertions: func(t *testing.T, err error, patched *argocd.Application) {
+				require.NoError(t, err)
+				require.NotNil(t, patched)
+				// Sources should be updated
+				require.Len(t, patched.Spec.Sources, 1)
+				require.Equal(t, "new-rev", patched.Spec.Sources[0].TargetRevision)
+				// Source should remain untouched
+				require.Equal(t, "old-rev-source", patched.Spec.Source.TargetRevision)
+			},
+		},
 	}
 
-	stepCtx := &promotion.StepContext{
-		Freight: kargoapi.FreightCollection{},
-	}
-	// Tamper with the freight collection ID for testing purposes
-	stepCtx.Freight.ID = "fake-freight-collection-id"
```

Comment:
You moved this up much higher and I'd prefer it remained here. It's a good practice to initialize things closer to where they are first used.

When you moved it, you also removed the important comment about the deliberate massaging of the ID.

Reply:
resolved.